server_addon/maya/client/ayon_maya/plugins/load/load_ornatrix_rig.py

```diff
@@ -0,0 +1,128 @@
+from typing import List
+
+import os
+import json
+import maya.cmds as cmds
+from ayon_core.pipeline import registered_host
+from ayon_core.pipeline.create import CreateContext
+from ayon_maya.api import lib, plugin
+
+
+class OxRigLoader(plugin.ReferenceLoader):
+    """This loader will load Ornatix rig."""
+
+    product_types = {"oxrig"}
+    representations = {"ma"}
+
+    label = "Load Ornatrix Rig"
+    order = -9
+    icon = "code-fork"
+    color = "orange"
+
+    # From settings
+    create_cache_instance_on_load = True
+
+    def process_reference(
+        self, context, name=None, namespace=None, options=None
+    ):
+        path = self.filepath_from_context(context)
+
+        # Check if the plugin for Ornatrix is available on the pc
+        try:
+            cmds.loadPlugin("Ornatrix.mll", quiet=True)
+        except Exception as exc:
+            self.log.error("Encountered exception:\n%s" % exc)
+            return
+
+        attach_to_root = options.get("attach_to_root", True)
+        group_name = options["group_name"]
+
+        # no group shall be created
+        if not attach_to_root:
+            group_name = namespace
+
+        with lib.maintained_selection():
+            file_url = self.prepare_root_value(
+                path, context["project"]["name"]
+            )
+            nodes = cmds.file(
+                file_url,
+                namespace=namespace,
+                reference=True,
+                returnNewNodes=True,
+                groupReference=attach_to_root,
+                groupName=group_name
+            )
+
+        color = plugin.get_load_color_for_product_type("oxrig")
+        if color is not None:
+            red, green, blue = color
+            cmds.setAttr(group_name + ".useOutlinerColor", 1)
+            cmds.setAttr(
+                group_name + ".outlinerColor", red, green, blue
+            )
+        self.use_resources_textures(namespace, path)
```

Comment:
Note that this would make "reference overrides" - preferably these textures are remapped before publishing on the extracted output file (similar to Look publish) so that you don't need 'fixing' on load.

Reply:
I implemented the contextmanager function of `attribute_values` inside the `extract_ornatrix_rig`